### PR TITLE
Change NodeID to NodeKey

### PR DESCRIPTION
**Pull request by zeina1i**
Hi @emilbon99 
Currently the NodeID is renamed to NodeKey in aspen and synnax directory.(this part ca be reviewed)
The left part is renaming calling them from clients. So please don't merge until this part is completed as well.

### CODE REVIEW

**Comment by zeina1i** [comment]
Hey @emilbon99

Client Test / test-typescript and Freighter Test / build-integration-image pipelines are failed.
But the weird thing is that when I trigger them in my local I don't have these errors. Do you have any idea?

**Comment by emilbon99** [comment]
> Hey @emilbon99
> 
> Client Test / test-typescript and Freighter Test / build-integration-image pipelines are failed. But the weird thing is that when I trigger them in my local I don't have these errors. Do you have any idea?

I'll debug. Give me an hour or two.

**Comment by emilbon99** [comment]
@zeina1i Issue fixed, just had to update the go mod file. The freighter tests aren't passing because PR's from forks don't have access to github packages. The client tests are failing because it's using the current production build of Synnax, which doesn't have your changes yet (I need to fix this :). What we can do for now is merge this PR into a separate branch on the main repo, make sure all CI passes, and then we should be good to go. Does that sound good?

**Comment by zeina1i** [comment]
@emilbon99 yes,
That sounds great👍

Should I do an action for that? Or you would do it?

**Comment by emilbon99** [comment]
I'll get it done.

File: aspen/internal/cluster/cluster.go [hunk 1]
```diff
@@ -59,15 +59,15 @@ type Cluster interface {
 
 // Resolver is used to resolve a reachable address for a node in the cluster.
 type Resolver interface {
-	// Resolve resolves the address of a node with the given ID.
-	Resolve(id node.ID) (address.Address, error)
+	// Resolve resolves the address of a node with the given Key.
+	Resolve(id node.Key) (address.Address, error)
```
Comment (emilbon99) [inline aspen/internal/cluster/cluster.go:63]: Can we change the `id` arg name to `key` as well?

File: aspen/internal/cluster/cluster.go [hunk 2]
```diff
@@ -167,14 +167,14 @@ func (c *cluster) Key() uuid.UUID {
...
-func (c *cluster) HostID() node.ID { return c.Store.PeekState().HostID }
+// HostKey implements the Cluster interface.
+func (c *cluster) HostKey() node.Key { return c.Store.PeekState().HostKey }
 
 // Nodes implements the Cluster interface.
 func (c *cluster) Nodes() node.Group { return c.Store.PeekState().Nodes }
 
 // Node implements the Cluster interface.
-func (c *cluster) Node(id node.ID) (node.Node, error) {
+func (c *cluster) Node(id node.Key) (node.Node, error) {
```
Comment (emilbon99) [inline aspen/internal/cluster/cluster.go:177]: See above. `id` -> `key`

File: aspen/internal/cluster/cluster.go [hunk 3]
```diff
@@ -183,7 +183,7 @@ func (c *cluster) Node(id node.ID) (node.Node, error) {
 }
 
 // Resolve implements the Cluster interface.
-func (c *cluster) Resolve(id node.ID) (address.Address, error) {
+func (c *cluster) Resolve(id node.Key) (address.Address, error) {
```
Comment (emilbon99) [inline aspen/internal/cluster/cluster.go:186]: `id` -> `key`

File: aspen/internal/cluster/pledge/pledge.go [hunk 4]
```diff
@@ -212,23 +212,23 @@ func (r *responsible) buildQuorum() (node.Group, error) {
...
 	} else {
-		r._proposedID++
+		r._proposedKey++
 	}
-	return r._proposedID
+	return r._proposedKey
 }
 
-func (r *responsible) consultQuorum(ctx context.Context, id node.ID, quorum node.Group) error {
+func (r *responsible) consultQuorum(ctx context.Context, id node.Key, quorum node.Group) error {
```
Comment (emilbon99) [inline aspen/internal/cluster/pledge/pledge.go:224]: `id` -> `key`

File: aspen/internal/cluster/pledge/pledge_test.go [hunk 5]
```diff
@@ -131,18 +131,18 @@ var _ = Describe("PledgeServer", func() {
...
+		Context("Responsible is Missing UniqueNodeKeys", func() {
+			It("Should correctly assign an Key", func() {
 				var (
 					nodes      = make(node.Group)
 					candidates = func(i int) func() node.Group {
 						return func() node.Group {
 							if i == 0 {
-								return nodes.Where(func(id node.ID, _ node.Node) bool {
-									return !lo.Contains([]node.ID{8, 9, 10}, id)
+								return nodes.Where(func(id node.Key, _ node.Node) bool {
```
Comment (emilbon99) [inline aspen/internal/cluster/pledge/pledge_test.go:144]: `id` -> `key`

File: aspen/internal/cluster/store/store.go [hunk 6]
```diff
@@ -108,43 +108,43 @@ func (c *core) SetClusterKey(key uuid.UUID) {
 }
 
 // GetNode implements Store.
-func (c *core) GetNode(id node.ID) (node.Node, bool) {
+func (c *core) GetNode(id node.Key) (node.Node, bool) {
```
Comment (emilbon99) [inline aspen/internal/cluster/store/store.go:111]: `id` -> `key`

File: aspen/internal/node/group.go [hunk 7]
```diff
@@ -14,20 +14,20 @@ import (
...
+type Group map[Key]Node
 
 func (n Group) WhereState(state State) Group {
-	return n.Where(func(_ ID, n Node) bool { return n.State == state })
+	return n.Where(func(_ Key, n Node) bool { return n.State == state })
 }
-func (n Group) WhereNot(ids ...ID) Group {
-	return n.Where(func(id ID, _ Node) bool { return lo.Count(ids, id) == 0 })
+func (n Group) WhereNot(ids ...Key) Group {
+	return n.Where(func(id Key, _ Node) bool { return lo.Count(ids, id) == 0 })
```
Comment (emilbon99) [inline aspen/internal/node/group.go:23]: `id` -> `key`

File: aspen/internal/node/group.go [hunk 8]
```diff
@@ -14,20 +14,20 @@ import (
...
-type Group map[ID]Node
+type Group map[Key]Node
 
 func (n Group) WhereState(state State) Group {
-	return n.Where(func(_ ID, n Node) bool { return n.State == state })
+	return n.Where(func(_ Key, n Node) bool { return n.State == state })
 }
-func (n Group) WhereNot(ids ...ID) Group {
-	return n.Where(func(id ID, _ Node) bool { return lo.Count(ids, id) == 0 })
+func (n Group) WhereNot(ids ...Key) Group {
```
Comment (emilbon99) [inline aspen/internal/node/group.go:22]: `ids` -> `keys`

File: aspen/internal/node/node.go [hunk 9]
```diff
@@ -10,26 +10,27 @@
...
-	"strconv"
 )
 
-type ID uint32
+type Key uint32
 
-func (i ID) Parse(str string) (ID, error) {
-	id, err := strconv.Atoi(str)
-	return ID(id), err
+func (i Key) Parse(str string) (Key, error) {
```
Comment (emilbon99) [inline aspen/internal/node/node.go:21]: `i` -> `k`

File: docs/rfc/2-220518-aspen-distributed-storage.mdx
Comment (emilbon99) [inline docs/rfc/2-220518-aspen-distributed-storage.mdx:1]: I generally like to keep RFCs frozen after they are completed to preserve a record of iteration. I think we should keep the term node ID in this RFC and instead add a section noting the change. What do you think?

File: docs/rfc/3-220604-segment-distribution.mdx
Comment (emilbon99) [inline docs/rfc/3-220604-segment-distribution.mdx:1]: See my comment on RFC 2.

File: docs/rfc/5-220716-ontology.mdx
Comment (emilbon99) [inline docs/rfc/5-220716-ontology.mdx:1]: The `ID` referenced in this RFC is unrelated to nodes. It should remain as `ID`.

File: synnax/pkg/distribution/channel/channel.go [hunk 10]
```diff
@@ -65,15 +65,15 @@ func ParseKey(s string) (k Key, err error) {
 	if len(split) != 2 {
 		return k, errors.New("[channel] - invalid key format")
 	}
-	nodeID, err := strconv.Atoi(split[0])
+	NodeKey, err := strconv.Atoi(split[0])
```
Comment (emilbon99) [inline synnax/pkg/distribution/channel/channel.go:68]: This should be camel case

File: synnax/pkg/distribution/core/ontology.go [hunk 11]
```diff
@@ -98,13 +98,13 @@ func (s *NodeOntologyService) RetrieveEntity(key string) (schema.Entity, error)
...
-	n, err := s.Cluster.Node(NodeID(id))
+	n, err := s.Cluster.Node(NodeKey(id))
 	return newNodeEntity(n), err
 }
 
 func newNodeEntity(n Node) schema.Entity {
-	e := schema.NewEntity(_nodeSchema, fmt.Sprintf("Node %v", n.ID))
-	schema.Set(e, "id", uint32(n.ID))
+	e := schema.NewEntity(_nodeSchema, fmt.Sprintf("Node %v", n.Key))
+	schema.Set(e, "id", uint32(n.Key))
```
Comment (emilbon99) [inline synnax/pkg/distribution/core/ontology.go:107]: `id` -> `key`

File: synnax/pkg/distribution/framer/writer/service.go [hunk 12]
```diff
@@ -158,7 +158,7 @@ func (s *Service) NewStream(ctx context.Context, cfg Config) (StreamWriter, erro
 	}
 
 	var (
-		hostID             = s.HostResolver.HostID()
+		hostID             = s.HostResolver.HostKey()
```
Comment (emilbon99) [inline synnax/pkg/distribution/framer/writer/service.go:161]: `hostID` -> `hostKey`

**Review by emilbon99 — CHANGES_REQUESTED** [review]
Thank you for doing all of this busy work! Aside from a few argument names that need to be changed and some RFC changes to rollback it looks great. Let me know when you've made these and we can get it merged into the main repo for CI.

File: docs/rfc/2-220518-aspen-distributed-storage.mdx
Comment (zeina1i) [reply docs/rfc/2-220518-aspen-distributed-storage.mdx:1]: I agree with you.
What do you think if we have an expand button infront of the part that we are intending the edit and when we click on expand a section would be opened which contains these items:
(in our case)
**original text**
Node.ID
**corrected text**
Node.Key
**notes**
A Node's ID uint32 ID is only unique to its subtype, and is not universally unique. So we rename Node.ID to Node.Key.


(If we don't have the expand button at the moment, We can also use an internal link to the "updates" section in the bottom of the page, but comparing to "expand button" it has the downside of the user has to return back to the original text after reading the updated section.)

Which one do you prefer?

File: docs/rfc/3-220604-segment-distribution.mdx
Comment (zeina1i) [reply docs/rfc/3-220604-segment-distribution.mdx:1]: See my comment on RFC 2 :)

File: docs/rfc/2-220518-aspen-distributed-storage.mdx
Comment (zeina1i) [reply docs/rfc/2-220518-aspen-distributed-storage.mdx:1]: Bytheway at the moment I rolledback the rfcs to their original version.
And we can proceed when we finalize our discusstion on how the update parts should be shown in the rfcs.

**Comment by zeina1i** [comment]
> Thank you for doing all of this busy work! Aside from a few argument names that need to be changed and some RFC changes to rollback it looks great. Let me know when you've made these and we can get it merged into the main repo for CI.

Thanks @emilbon99 
I resolved the namings.
And waiting on your opinion about the suggestions I had on the [rfcs](https://github.com/synnaxlabs/synnax/pull/236#discussion_r1155340233), Then I would do rfc changes as well.

**Comment by emilbon99** [comment]
@zeina1i I like the idea of adding a tooltip. I'll add that as another issue we can work in the future. Leaving the exiting version is fine for now.

**Comment by zeina1i** [comment]
@emilbon99 Good :),  so, then should I resolve the rfc discussions in this state?

**Pull request merged**